src/function/function.cpp

```diff
@@ -488,6 +488,19 @@ unique_ptr<BoundFunctionExpression> ScalarFunction::BindScalarFunction(ClientCon
 
 	// found a matching function!
 	auto &bound_function = func.functions[best_function];
+
+	if (bound_function.null_handling == NULL_IN_NULL_OUT) {
+		for (auto &child : children) {
+			if (child->return_type == LogicalTypeId::SQLNULL) {
+				// mark the result, telling the binder that it can be replaced with a NULL
+				auto result = make_unique<BoundFunctionExpression>(LogicalType::SQLNULL, bound_function, move(children),
+				                                                   nullptr, is_operator);
+				result->alias = ExpressionBinder::REPLACE_WITH_NULL;
```

Comment:
Can't we directly return a `BoundConstantExpression` with `Value(LogicalType::SQLNULL)` here? Using the alias for this is not pretty. It also seems like it might be a potential source of bugs, as this now returns an incomplete function that might crash when actually used as a function (since the bind_data is not set).

Reply:
I agree that it's not great. However, we don't have access to the binder there, so we can't remove the parameter expressions in there when we find out the result has to be NULL. We also can't return a BoundConstantExpression because we have to return a BoundFunctionExpression.

We could add the binder as a parameter, and then do the logic within BindScalarFunction, and then we can return a NOP BoundFunctionExpression that returns NULL, e.g, +(NULL). What do you think?

Reply:
Can we not change the return type of this function to be a generic expression?